presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarballController.java

```diff
@@ -0,0 +1,138 @@
+import static com.teradata.prestomanager.agent.PrestoConfigUtils.untarFile;
+import static java.lang.String.format;
+import static java.nio.file.Files.createDirectories;
+import static java.nio.file.Files.isDirectory;
+
+public class TarballController
+        extends PackageController
+{
+    private static final Logger LOGGER = Logger.get(TarballController.class);
+
+    private static final Path TAR_INSTALLATION_DIR = Paths.get("/usr/presto");
+    private static final Path TAR_DATA_DIR = Paths.get("/usr/data");
+
+    public void installAsync(URL packageUrl, boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        File tarTempFile = getPackageFile(packageUrl);
+        try {
+            tarInstall(tarTempFile);
+            addConfigFiles(CONFIG_DIR);
+            addConnectors(CATALOG_DIR);
+        }
+        finally {
+            if (!tarTempFile.delete()) {
+                LOGGER.warn("Failed to delete the temporary file: %s", tarTempFile.toString());
+            }
+        }
+        LOGGER.debug("Successfully installed Presto");
+    }
+
+    private void tarInstall(File tarFile)
+            throws PrestoManagerException
+    {
+        try {
+            if (isDirectory(TAR_INSTALLATION_DIR)) {
+                throw new PrestoManagerException(format("Directory '%s' already exist", TAR_INSTALLATION_DIR.toString()));
+            }
+            else {
+                createDirectories(TAR_INSTALLATION_DIR);
+            }
+            untarFile(tarFile, TAR_INSTALLATION_DIR);
+            createDirectories(CONFIG_DIR);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to install Presto");
+        }
+    }
+
+    public void uninstallAsync(boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        try {
+            deleteRecursively(TAR_INSTALLATION_DIR);
+            deleteRecursively(TAR_DATA_DIR);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to uninstall Presto", e);
+        }
+        LOGGER.debug("Successfully uninstalled Presto");
+    }
+
+    public void upgradePackage(File tarFile, boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        uninstallAsync(checkDependencies);
+        tarInstall(tarFile);
+    }
+
+    public File getPackageFile(URL packageUrl)
+            throws PrestoManagerException
+    {
+        File tarTempFile;
+        try {
+            LOGGER.debug("Downloading file from url: %s", packageUrl);
+            tarTempFile = downloadFile(packageUrl).toFile();
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException(format("Failed to download the file from url : %s", packageUrl.toString()));
+        }
+        return tarTempFile;
+    }
+
+    public void startAsync()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
```

Comment:
If these are temporary placeholders, they should throw `UnsupportedOperationException("Not implemented")`, and have TODOs to implement.

Also, the implementations of abstract methods should be annotated with `@Overrides`.
I forgot to say this on `RpmController`, but they should also be annotated there.